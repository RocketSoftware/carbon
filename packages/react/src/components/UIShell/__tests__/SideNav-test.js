/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import SideNav from '../SideNav';

describe('SideNav', () => {
  let mockProps, wrapper;

  beforeEach(() => {
    mockProps = {
      'aria-label': 'Navigation',
      children: <h2>Navigation</h2>,
    };
  });

  afterEach(() => {
    wrapper && wrapper.unmount();
  });

  it('should render', () => {
    wrapper = mount(<SideNav {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('by default, all event listeners are added', () => {
    wrapper = mount(<SideNav {...mockProps} />);
    expect(wrapper.find('nav').props().onFocus).toBeDefined();
    expect(wrapper.find('nav').props().onBlur).toBeDefined();
    expect(wrapper.find('nav').props().onMouseEnter).toBeDefined();
    expect(wrapper.find('nav').props().onMouseLeave).toBeDefined();
  });

  it('if addFocusListeners is specified as false, no focus event listeners props are added', () => {
    wrapper = mount(<SideNav {...mockProps} />);
    wrapper.setProps({ addFocusListeners: false });
    expect(wrapper.find('nav').props().onFocus).not.toBeDefined();
    expect(wrapper.find('nav').props().onBlur).not.toBeDefined();
    expect(wrapper.find('nav').props().onMouseEnter).toBeDefined();
    expect(wrapper.find('nav').props().onMouseLeave).toBeDefined();
  });

  it('if addMouseListeners is specified as false, no mouse listener props are added', () => {
    wrapper = mount(<SideNav {...mockProps} />);
    wrapper.setProps({ addMouseListeners: false });
    expect(wrapper.find('nav').props().onFocus).toBeDefined();
    expect(wrapper.find('nav').props().onBlur).toBeDefined();
    expect(wrapper.find('nav').props().onMouseEnter).not.toBeDefined();
    expect(wrapper.find('nav').props().onMouseLeave).not.toBeDefined();
  });

  it('if both addFocusListeners and addMouseListeners are specified as false, no mouse or focus listener props are added', () => {
    wrapper = mount(<SideNav {...mockProps} />);
    wrapper.setProps({ addFocusListeners: false, addMouseListeners: false });
    expect(wrapper.find('nav').props().onFocus).not.toBeDefined();
    expect(wrapper.find('nav').props().onBlur).not.toBeDefined();
    expect(wrapper.find('nav').props().onMouseEnter).not.toBeDefined();
    expect(wrapper.find('nav').props().onMouseLeave).not.toBeDefined();
  });
});
