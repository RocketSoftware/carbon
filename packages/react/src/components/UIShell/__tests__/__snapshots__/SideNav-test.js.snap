// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render 1`] = `
<ForwardRef(SideNav)
  addFocusListeners={true}
  addMouseListeners={true}
  aria-label="Navigation"
  defaultExpanded={false}
  isChildOfHeader={true}
  isFixedNav={false}
  isPersistent={true}
  translateById={[Function]}
>
  <div
    className="bx--side-nav__overlay"
  />
  <nav
    aria-label="Navigation"
    className="bx--side-nav__navigation bx--side-nav bx--side-nav--ux"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <h2>
      Navigation
    </h2>
  </nav>
</ForwardRef(SideNav)>
`;
